### PR TITLE
doc/1132_add_integration_examples

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #1132 
* No tests are required
* Assignee: @frbattid

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -5,3 +5,4 @@
 - [cygnus-common][hardening] Create ManagementInterfaceUtils class (#1115)
 - [cygnus-common][bug] Add fiware-service and fiware-servicePath headers to all API operations (#1124)
 - [cygnus-ngsi][hardening] Docker image must print the logs in console (#1125)
+- [cygnus][doc] Add integration path and Orion-Cygnus-Kafka example (#1132)
```
Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: Add Orion-Cygnus-Kafka integration example

File: README.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ Current stable release is able to persist the following sources of data in the f
 ##Further documentation
 The per agent **Quick Start Guide** provides a good documentation summary.
 
-Nevertheless, both the **Installation and Administration Guide** and the **User and Programmer Guide** shown within this documentation for each agent cover more advanced topics.
+Nevertheless, both the **Installation and Administration Guide** and the **User and Programmer Guide** shown within this documentation for each agent cover more advanced topics. Of special interest are the per-agent integration examples under `doc/<agent>/integration`.
```
Comment (frbattid) [inline README.md]: Use a new paragraph.

File: doc/index.md [hunk 3]
```diff
@@ -34,7 +34,7 @@ Current stable release is able to persist the following sources of data in the f
 ##About this documentation
 The per agent **Quick Start Guide** provides a good documentation summary.
 
-Nevertheless, both the **Installation and Administration Guide** and the **User and Programmer Guide** shown within this documentation for each agent cover more advanced topics.
+Nevertheless, both the **Installation and Administration Guide** and the **User and Programmer Guide** shown within this documentation for each agent cover more advanced topics. Of special interest are the per-agent integration examples under `doc/<agent>/integration`.
```
Comment (frbattid) [inline doc/index.md]: Use a new paragraph.

Comment (pcoello25) [inline doc/index.md]: Fixed in  bb93dba

File: README.md (hunk 2 above)
Comment (pcoello25) [inline README.md]: Fixed in  bb93dba

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (pcoello25) [inline CHANGES_NEXT_RELEASE]: Fixed in  bb93dba

File: doc/cygnus-ngsi/integration/orion_cygnus_kafka.md [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+[2016-08-30 07:33:39,337] INFO minSessionTimeout set to -1 (org.apache.zookeeper.server.ZooKeeperServer)
+[2016-08-30 07:33:39,337] INFO maxSessionTimeout set to -1 (org.apache.zookeeper.server.ZooKeeperServer)
+[2016-08-30 07:33:39,446] INFO binding to port 0.0.0.0/0.0.0.0:2181 (org.apache.zookeeper.server.NIOServerCnxnFactory)
+[2016-08-30 07:33:54,325] INFO Accepted socket connection from /0:0:0:0:0:0:0:1:45268 (org.apache.zookeeper.server.NIOServerCnxnFactory)
+[2016-08-30 07:33:54,412] INFO Client attempting to establish new session at /0:0:0:0:0:0:0:1:45268 (org.apache.zookeeper.server.ZooKeeperServer)
+[2016-08-30 07:33:54,415] INFO Creating new log file: log.1 (org.apache.zookeeper.server.persistence.FileTxnLog)
+[2016-08-30 07:33:54,485] INFO Established session 0x156d9f04d320000 with negotiated timeout 6000 for client /0:0:0:0:0:0:0:1:45268 (org.apache.zookeeper.server.ZooKeeperServer)
+```
+
+As we see in the image the execution of this command use the entire console. At this moment we have two options:
```
Comment (frbattid) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: As we can see in the text above... (there is not an image :))

File: doc/cygnus-ngsi/integration/orion_cygnus_kafka.md [hunk 5]
```diff
@@ -0,0 +1,756 @@
...
+[2016-08-30 07:33:55,008] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
+[2016-08-30 07:33:55,009] INFO [Socket Server on Broker 1], Started (kafka.network.SocketServer)
+[2016-08-30 07:33:55,254] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
+[2016-08-30 07:33:55,314] INFO 1 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
+[2016-08-30 07:33:55,438] INFO Registered broker 1 at path /brokers/ids/1 with address 0.0.0.0:9092. (kafka.utils.ZkUtils$)
+[2016-08-30 07:33:55,511] INFO [Kafka Server 1], started (kafka.server.KafkaServer)
+[2016-08-30 07:33:55,603] INFO New leader is 1 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
+```
+
+As we see in the image the execution of this command use the entire console. At this moment we have two options:
```
Comment (frbattid) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: As we can see in the text above...

File: doc/cygnus-ngsi/integration/orion_cygnus_kafka.md [hunk 6]
```diff
@@ -0,0 +1,756 @@
...
+time=2016-08-30T08:03:01.063CEST | lvl=INFO | corr=31ce961a-2767-4acd-bd5e-b623c3062148 | trans=31ce961a-2767-4acd-bd5e-b623c3062148 | srv=library | subsrv=/catalog | function=processNewBatches | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.NGSISink[400] : Finishing internal transaction (31ce961a-2767-4acd-bd5e-b623c3062148)
+time=2016-08-30T08:03:16.065CEST | lvl=INFO | corr=31ce961a-2767-4acd-bd5e-b623c3062148 | trans=31ce961a-2767-4acd-bd5e-b623c3062148 | srv=library | subsrv=/catalog | function=processNewBatches | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.NGSISink[342] : Batch accumulation time reached, the batch will be processed as it is
+time=2016-08-30T08:03:16.065CEST | lvl=INFO | corr=31ce961a-2767-4acd-bd5e-b623c3062148 | trans=31ce961a-2767-4acd-bd5e-b623c3062148 | srv=library | subsrv=/catalog | function=processNewBatches | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.NGSISink[400] : Finishing internal transaction ()
+```
+
+[Top](#top)
+
+##<a name="section7"></a>Updating subscription
```
Comment (frbattid) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: Updating entities

Comment (frbattid) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: Check the index as well.

File: doc/cygnus-ngsi/integration/orion_cygnus_kafka.md [hunk 7]
```diff
@@ -0,0 +1,756 @@
...
+* [Introduction](#section1)
+* [Running Orion](#section2)
+* [Running Kafka](#section3)
+    * [Zookeeper](#section3.1)
+    * [Brokers](#section3.2)
+* [Running Cygnus](#section4)
+* [Creating a subscription](#section5)
+* [Appending entities](#section6)
+* [Updating subscription](#section7)
+* [Conclusion](#section8)
```
Comment (frbattid) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: I think this section does not exist.

File: doc/cygnus-ngsi/integration/orion_cygnus_kafka.md (hunk 4 above)
Comment (pcoello25) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: Oops, my fault, forgot to modify the reference :fearful: 
Fixed in 00ae546

File: doc/cygnus-ngsi/integration/orion_cygnus_kafka.md (hunk 5 above)
Comment (pcoello25) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: Fixed in 00ae546

File: doc/cygnus-ngsi/integration/orion_cygnus_kafka.md (hunk 6 above)
Comment (pcoello25) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: Index is right, there are another inexistent section (your next comment).
Fixed in 00ae546

File: doc/cygnus-ngsi/integration/orion_cygnus_kafka.md (hunk 7 above)
Comment (pcoello25) [inline doc/cygnus-ngsi/integration/orion_cygnus_kafka.md]: Yep, It existed in the past :laughing: 
Fixed in 00ae546

**Comment by frbattid** [comment]
LGTM

**Pull request merged**